doc/Language/rb-nutshell.pod

```diff
@@ -0,0 +1,944 @@
+done using the C<|> prefix:
+
+    args = %w(a b c)         # Ruby
+    foo(*args)
+
+    my @args = <a b c>       # Perl 6
+    foo(|@args)
+
+=head3 Twigils
+
+Perl 6 additionally uses "twigils", which are further indicators about the
+variable and go between the sigil and the rest of the variable name. Examples:
+
+    $foo     # Scalar with no twigil
+    $!foo    # Private instance variable
+    $.foo    # Instance variable accessor
+    $*foo    # Dynamically scoped variable
+    $^foo    # A positional (placeholder) parameter to a block
+    $:foo    # A named parameter
+    $=foo    # POD (documentation) variables
+    $?FILE   # Current source filename. The ? twigil indicates a compile-time value
+    $~foo    # Sublanguage seen by parser, uncommon
+
+=head3 [] Array indexing/slicing
+
+Index and slice operations on arrays no longer inflect the variable's
+L<sigil|#@_Array>, and adverbs can be used to control the type of slice:
+
+=begin item
+I<Indexing>
+
+    puts months[2]; # Ruby
+    say @months[2]; # Perl 6 - @ instead of $
+=end item
+
+=begin item
+I<Value-slicing>
+
+    puts months[8..11].join(',') # Ruby
+    say @months[8..11].join(',') # Perl 6
+=end item
+
+Also note that the subscripting brackets are a normal postcircumfix operator
+rather than a special syntactic form, and thus L<checking for existence of
+elements|#exists> and L<unsetting elements|#delete> is done with adverbs.
+
+=head3 {} Hash indexing/slicing
+
+Ruby uses square brackets to access values for both Arrays and Hashes. Perl 6
+uses curley braces for hashes instead. The angle-brackets version is available
+which always autoquotes its contents (strings without quotes):
+
+Adverbs can be used to control the type of slice.
+
+=begin item
+I<Indexing>
+
+    puts calories["apple"]  # Ruby
+    say %calories{"apple"}; # Perl 6 - % instead of $
+
+    puts calories["apple"]  # Ruby
+    puts calories[:apple]   # Ruby, symbols for keys are common
+    say %calories<apple>;   # Perl 6 - angle brackets instead of single-quotes
+    say %calories«$key»;    # Perl 6 - double angles interpolate as double-quotes
+=end item
+
+=begin item
+I<Value-slicing>
+
+    puts calories.values_at('pear', 'plum').join(',') # Ruby
+    puts calories.values_at(%w(pear plum)).join(',')  # Ruby, pretty?
+    say %calories{'pear', 'plum'}.join(',');          # Perl 6
+    say %calories<pear plum>.join(',');               # Perl 6 (prettier)
+    my $keys = 'pear plum';
+    say %calories«$keys».join(','); # Perl 6, interpolated split
+=end item
+
+=begin item
+I<Key/value-slicing>
+
+    say calories.slice('pear', 'plum').join(','); # Ruby, with ActiveRecord
+    say %calories{'pear', 'plum'}:kv.join(',');   # Perl 6 - use :kv adverb
+    say %calories<pear plum>:kv.join(',');        # Perl 6 (prettier version)
+=end item
+
+Also note that the subscripting curly braces are a postcircumfix operator
+rather than a special syntactic form, and thus L<checking for existence of
```

Comment:
Again, not sure that is totally true, and provide an example of existence and removal adverbs